blog/2023-03-14-introduction-to-erc-20r-and-reversible-transactions/introduction-to-erc-20r-and-reversible-transactions.md

```diff
@@ -0,0 +1,87 @@
+
+- **freeze()**
+
+This function calculates and freezes the amounts to be frozen on the attacker's address and potential downstream addresses. It returns a claimID for ERC-20R that points to an on-chain list of account (amount) pairs. The list identifies all of the accounts containing frozen assets related to the complaint, as well as the amount frozen in each. It returns a boolean success flag for ERC-721R. Section 2 explains the inner workings of the freeze function.
+
+- **reverse()**
+
+All frozen assets associated with the claimed theft are returned to the original owner using this function. It takes a valid claimID as an argument for ERC-20R and an argument (tokenId; index) for ERC-721R, where the index identifies the reversed transaction.
+
+- **rejectReverse():** All amounts associated with the claim are unfrozen using the rejectReverse() function. It takes a claimID as an argument for ERC-20R and a tokenId for ERC-721R.
+
+- **clean():** Some transaction data is stored on the chain by reversible contracts. The clean function deletes on-chain information for transactions with a closed dispute window.
+
+_**Side Note:** The freeze(), reverse(), and rejectreverse() functions can only be called by the governance contract, while anyone can call the clean function._
+
+## Real-Life Implementation
+
+The full ERC20R contract has been implemented by Kaili Wang, which can be found [here](https://github.com/kkailiwang/erc20r/blob/master/contracts/ERC20R.sol).
+
+## Conclusion
+
+With intensive research and development, reversible transactions are rapidly evolving. When fully operational, they will contribute significantly to the exponential growth of the crypto industry and reduce hacks and thefts, thereby increasing consumer trust and confidence. This type of technology has a lot of advantages, especially for crypto institutions, because it adds an extra layer of security and allows them to avoid being stuck with an irreversible payment without their consent.
+
+## References
+
+- Kaili Wang, Dan Boneh, and Qinchen Wang (October 2022), ERC-20 and ERC-721: Reversible Transactions on Ethereum. Available at [https://arxiv.org/abs/2208.00543](https://arxiv.org/abs/2208.00543)
+- Kaili Wang, (September 2022), Reversible Transactions on Ethereum: ERC-20R and ERC-721R Available at [https://mirror.xyz/kaili.eth/gB-rx89sNAT3CVuxWo6xVFS5ptNcllW7cVWVCfcFa6k](https://mirror.xyz/kaili.eth/gB-rx89sNAT3CVuxWo6xVFS5ptNcllW7cVWVCfcFa6k)
+
+## Author
+
+[Oyeniyi Abiola Peace](https://twitter.com/_iamoracle) is a seasoned software and blockchain developer. With a degree in Telecommunication Science from the University of Ilorin and over five years of experience in JavaScript, Python, PHP, and Solidity, he is no stranger to the tech industry. Peace currently works as the CTO at DFMLab and is a Community Moderator at Celo Blockchain. When he's not coding or teaching, he loves to read and spend time with family and friends.
```

Comment:
Removing twitter id, as we are not supposed to add that, you can use github or linkdln

Reply:
This time , I am resolving it ,make sure it does not happen in future